### PR TITLE
3.0.0 Release

### PR DESCRIPTION
**Pull request by nblumhardt**
- #131 - move to _Microsoft.Extensions.Logging_ v2.0 dependencies throughout, drop 1.x .NET Core versions (@DigiBanks99)
 - #132 - `SerilogLoggerFactory` and `AddProvider()` support (@nblumhardt)
 - #138 - move to current SDK (@SimonCropp)
 - #140 - remove unused build-time constants (@nblumhardt)
 - #141 - make `LevelConvert` level-mapping routines public
 - #148 - build with VS2019 (@nblumhardt)
 - #142 - (partially) cache low-numbered event ids (@nblumhardt)

### CODE REVIEW

**Comment by nblumhardt** [comment]
Folks, any thoughts/feelings/perspectives we're missing on whether this should go out the door as-is? We've had the brakes on in case there were breaking changes in the ASP.NET Core 3.x series, but it doesn't seem like that's going to be the case, so we should probably get this all out in the wild :-)

**Comment by nblumhardt** [comment]
Ideally before merging this we'll finish #142 and #148

File: appveyor.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 version: '{build}'
 skip_tags: true
-image: Visual Studio 2017 Preview
+image: Visual Studio 2019 Preview
```
Comment (adamchester) [inline appveyor.yml]: Are we choosing preview on purpose?

Comment (adamchester) [reply appveyor.yml]: No we are not. As I saw in #148 😁

**Comment by nblumhardt** [comment]
#142 is ready to go

**Pull request merged**